src/utils.js

```diff
@@ -619,35 +619,58 @@ export function shouldRender(comp, nextProps, nextState) {
   return !deepEquals(props, nextProps) || !deepEquals(state, nextState);
 }
 
-export function toIdSchema(
-  schema,
-  id,
-  definitions,
-  formData = {},
-  idPrefix = "root"
-) {
+export function toIdSchema(schema, id, definitions, formData, seperator = "_") {
   const idSchema = {
-    $id: id || idPrefix,
+    $id: id || "root",
   };
-  if ("$ref" in schema) {
+  if ("$ref" in schema || "dependencies" in schema) {
     const _schema = retrieveSchema(schema, definitions, formData);
-    return toIdSchema(_schema, id, definitions, formData, idPrefix);
-  }
-  if ("items" in schema && !schema.items.$ref) {
-    return toIdSchema(schema.items, id, definitions, formData, idPrefix);
+    return toIdSchema(_schema, id, definitions, formData, seperator);
+  }
+  if ("items" in schema) {
+    for (const i in formData) {
+      const itemId = idSchema.$id + seperator + i;
+      if (Array.isArray(schema.items)) {
+        if (i < schema.items.length) {
+          idSchema[i] = toIdSchema(
+            schema.items[i],
+            itemId,
+            definitions,
+            formData[i]
+          );
+        } else if (schema.additionalItems) {
+          idSchema[i] = toIdSchema(
+            schema.additionalItems,
+            itemId,
+            definitions,
+            formData[i]
+          );
+        }
+        // exception: formData is longer the the given length
```

Comment:
Do we have a preferred behavior for this case?

Reply:
This happen when the schema is a fix length array but the form data is longer then it. If those exceed are not go be displayed on the form then it should be fine not assigning an id.